src/commands/createFunctionApp/FunctionAppCreateStep.ts

```diff
@@ -83,46 +84,22 @@ export class FunctionAppCreateStep extends AzureWizardExecuteStep<IFunctionAppWi
     /**
      * Has a few temporary workarounds so that the sdk allows some newer properties on the plan
      */
-    private addCustomLocationProperties(site: WebSiteManagementModels.Site, customLocation: CustomLocation): void {
+    private addCustomLocationProperties(site: Site, customLocation: CustomLocation): void {
         nonNullProp(site, 'siteConfig').alwaysOn = true;
-
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        WebSiteManagementMappers.Site.type.modelProperties!.extendedLocation = {
-            serializedName: 'extendedLocation',
-            type: {
-                name: "Composite",
-                modelProperties: {
-                    name: {
-                        serializedName: "name",
-                        type: {
-                            name: "String"
-                        }
-                    },
-                    type: {
-                        serializedName: "type",
-                        type: {
-                            name: "String"
-                        }
-                    }
-                }
-            }
-        };
-
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
-        (<any>site).extendedLocation = { name: customLocation.id, type: 'customLocation' };
+        site.extendedLocation = { name: customLocation.id, type: 'customLocation' };
```

Comment:
Similar to Brandon's PR with the appservice package, this mapper was used as a workaround to allow some newer properties without actually having to upgrade.  

https://github.com/microsoft/vscode-azuretools/pull/1061/files#r802981607